File: meetup/client/src/api/responses.ts
import { Meet, User } from "../types";

export interface BaseResponse {
	msg: string;
	success: boolean;
	status_code: number;
	payload: object;
}

export interface CurrentActiveUserResponse extends BaseResponse {
	payload: CurrentActiveUserPayload;
}

export interface UsersResponse extends BaseResponse {
	payload: UsersPayload;
}

export interface UserResponse extends BaseResponse {
	payload: UserPayload;
}

export interface MeetsResponse extends BaseResponse {
	payload: MeetsPayload;
}
export interface MeetResponse extends BaseResponse {
	payload: MeetPayload;
}

export interface MeetPayload {
	meet: Meet;
}

export interface MeetsPayload {
	meets?: Meet[];
}

export interface CurrentActiveUserPayload {
	user?: User;
	authenticated: boolean;
}

export interface UsersPayload {
	users: User[];
}

export interface UserPayload {
	user: User;
}

File: meetup/server/app/services/users/responses.py
from pydantic import BaseModel

from app.services.responses import BaseResponse
from .schemas import UserSchema


class CurrentActiveUserResponse(BaseResponse):
    class Payload(BaseModel):
        user: UserSchema
        authenticated: bool

    payload: Payload


class Users(BaseResponse):
    class Payload(BaseModel):
        users: list[UserSchema]

    payload: Payload


class User(BaseResponse):
    class Payload(BaseModel):
        user: UserSchema

    payload: Payload
